pkg/baseconn/conn.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package baseconn
+
+import (
+	"database/sql"
+	tcontext "github.com/pingcap/dm/pkg/context"
+	"github.com/pingcap/dm/pkg/log"
+	"github.com/pingcap/dm/pkg/retry"
+	"github.com/pingcap/dm/pkg/terror"
+
+	"go.uber.org/zap"
+)
+
+// BaseConn wraps a connection to DB
+type BaseConn struct {
+	DB *sql.DB
+
+	// for reset
+	DSN string
+
+	RetryStrategy retry.Strategy
+}
+
+// SQL is format sql job
+type SQL struct {
+	Query string
+	Args  []interface{}
+}
+
+// NewBaseConn builds BaseConn to connect real DB
+func NewBaseConn(dbDSN string) (*BaseConn, error) {
+	db, err := sql.Open("mysql", dbDSN)
+	if err != nil {
+		return nil, terror.ErrDBDriverError.Delegate(err)
+	}
+	err = db.Ping()
+	if err != nil {
+		return nil, terror.ErrDBDriverError.Delegate(err)
+	}
+	return &BaseConn{db, dbDSN, nil}, nil
+}
+
+// SetRetryStrategy set retry strategy for baseConn
+func (conn *BaseConn) SetRetryStrategy(strategy retry.Strategy) error {
+	if conn == nil {
+		return terror.ErrDBDriverError.Generate("no valid connection")
```

Comment:
how about using `terror.ErrDBUnExpect.Generate("database connection not valid")` as we did in other places (also in L67)?